static/js/src/advantage/subscribe/react/APICalls/PendingPurchase.jsx

```diff
@@ -0,0 +1,77 @@
+import { useStripe } from "@stripe/react-stripe-js";
+import { useState } from "react";
+import { useQuery } from "react-query";
+import { getPurchase, postInvoiceID } from "../../../contracts-api";
+
+const usePendingPurchase = () => {
+  const [pendingPurchaseID, setPendingPurchaseID] = useState();
+
+  const stripe = useStripe();
+
+  const { isLoading, isError, isSuccess, data, error } = useQuery(
+    "pendingPurchase",
+    async () => {
+      const res = await getPurchase(pendingPurchaseID);
+
+      if (!res.stripeInvoices) {
+        throw new Error("Missing invoice");
+      }
+
+      const {
+        pi_decline_code,
+        pi_status,
+        pi_secret,
+        id: stripeInvoiceId,
+      } = res.stripeInvoices[0];
+
+      //Requires 3DS check
+      if (
+        pi_decline_code === "authentication_required" ||
+        (pi_status === "requires_action" && pi_secret)
+      ) {
```

Comment:
You could move this conditional to a function called `if (requires3DSCheck(..)) {` which would explain itself without the need for the comment.